gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/repository/mapper/ApiMapper.java

```diff
@@ -62,10 +64,17 @@ public Maybe<ReactableApi<?>> to(Event apiEvent) {
                     // Update definition with required information for deployment phase
                     reactableApi = new io.gravitee.gateway.handlers.api.definition.Api(eventApiDefinition);
                 } else {
-                    var eventApiDefinition = objectMapper.readValue(api.getDefinition(), io.gravitee.definition.model.v4.Api.class);
+                    if (api.getType() == ApiType.NATIVE) {
+                        var eventApiDefinition = objectMapper.readValue(api.getDefinition(), NativeApi.class);
 
-                    // Update definition with required information for deployment phase
-                    reactableApi = new io.gravitee.gateway.reactive.handlers.api.v4.Api(eventApiDefinition);
+                        // Update definition with required information for deployment phase
+                        reactableApi = new io.gravitee.gateway.reactive.handlers.api.v4.NativeApi(eventApiDefinition);
+                    } else {
```

Comment:
Maybe we can also check the `PROXY` or `MESSAGE` type and add something `else` that throws an IllegalArgumentException("Unsupported ApiType ...")?